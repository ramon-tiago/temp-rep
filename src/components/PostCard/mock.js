export const postCardPropsMock = {
  title: 'Hello',
  body: 'body 1',
  id: '1',
  cover: 'img/img.png',
};