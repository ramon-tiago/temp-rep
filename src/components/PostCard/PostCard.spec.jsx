import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard>', () => {
  it('Should render PostCard correctly', () => {
    render(<PostCard post={props} />);
    expect(screen.getByRole('img', { name: /Hello/i })).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: 'Hello' })).toBeInTheDocument();
    expect(screen.getByText('body 1')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard post={props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
