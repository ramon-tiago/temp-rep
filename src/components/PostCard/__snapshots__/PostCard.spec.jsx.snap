// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Hello"
    class="cover"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      Hello
    </h1>
    <p>
      body 1
    </p>
  </div>
</div>
`;
